File: app/src/main/java/ai/elimu/appstore/room/entity/Application.java
package ai.elimu.appstore.room.entity;

import androidx.annotation.NonNull;
import androidx.room.Entity;

import java.util.Set;

import ai.elimu.model.enums.admin.ApplicationStatus;
import ai.elimu.model.enums.content.LiteracySkill;
import ai.elimu.model.enums.content.NumeracySkill;

/**
 * For documentation, see https://github.com/elimu-ai/webapp/tree/master/src/main/java/ai/elimu/model
 */
@Entity
public class Application extends BaseEntity {

    @NonNull
    private String packageName;

    private Boolean infrastructural;

    @NonNull
    private ApplicationStatus applicationStatus;

    private Set<LiteracySkill> literacySkills;

    private Set<NumeracySkill> numeracySkills;

    public String getPackageName() {
        return packageName;
    }

    public void setPackageName(String packageName) {
        this.packageName = packageName;
    }

    public Boolean getInfrastructural() {
        return infrastructural;
    }

    public void setInfrastructural(Boolean infrastructural) {
        this.infrastructural = infrastructural;
    }

    public ApplicationStatus getApplicationStatus() {
        return applicationStatus;
    }

    public void setApplicationStatus(ApplicationStatus applicationStatus) {
        this.applicationStatus = applicationStatus;
    }

    public Set<LiteracySkill> getLiteracySkills() {
        return literacySkills;
    }

    public void setLiteracySkills(Set<LiteracySkill> literacySkills) {
        this.literacySkills = literacySkills;
    }

    public Set<NumeracySkill> getNumeracySkills() {
        return numeracySkills;
    }

    public void setNumeracySkills(Set<NumeracySkill> numeracySkills) {
        this.numeracySkills = numeracySkills;
    }
}

File: app/src/main/java/ai/elimu/appstore/room/entity/Application.kt
package ai.elimu.appstore.room.entity

import ai.elimu.model.v2.enums.admin.ApplicationStatus
import ai.elimu.model.v2.enums.content.LiteracySkill
import ai.elimu.model.v2.enums.content.NumeracySkill
import androidx.room.Entity

/**
 * For documentation, see https://github.com/elimu-ai/webapp/tree/master/src/main/java/ai/elimu/model
 */
@Entity
class Application : BaseEntity() {
    lateinit var packageName: String

    @JvmField
    var infrastructural: Boolean? = null

    lateinit var applicationStatus: ApplicationStatus

    @JvmField
    var literacySkills: Set<LiteracySkill>? = null

    @JvmField
    var numeracySkills: Set<NumeracySkill>? = null
}
